### DISCUSSION TITLE
理想的 IM APP

### DISCUSSION THREAD

**Discussion by kouhe3** [kouhe3/discussion#6] [Ideas]
常用的 IM APP 有很多不爽的地方。把这些不爽的地方整理到一起，也许可以预见理想的 IM APP 是什么样子的。

**Comment by kouhe3** [comment]
tg 会莫名其妙封号，qq 会封群。
IM 可以去中心化就好了，可以自己托管服务器。
但是如果不能跨域通信，那就成了孤岛了。
所以可以自己搭建服务器，又可以跨服务器通信才好。
参考 matrix 和 xmpp

**Comment by kouhe3** [comment]
聊天记录的检索也很重要，如果历史聊天记录不能被搜索，那就太蛋疼了。
然后聊天记录的迁移，导出，导入，抹除，本地与云端。相关的功能也得做得好。

**Comment by kouhe3** [comment]
然后就是加密。可以用 PGP 或者别的什么验证身份的话更好。

**Comment by kouhe3** [comment]
扩展性也很重要啊，还有互联。
用一个 IM APP 接收另一个平台的消息是很棒的。
点名 Pidgin 

还有机器人功能，
一个 IM 上的机器人强不强大也很重要

**Comment by kouhe3** [comment]
人们可能想分主题讨论，大部分这样的需求都让论坛解决了。
如果 IM APP 可以开小分区针对一个主题讨论就好了

**Comment by kouhe3** [comment]
应该爬楼友好——即可以记住上次看到哪里了，下次接着看。翻阅未读消息的时候应该从第一条开始，而不是最新的开始。

**Comment by Nine499** [comment]
呀哈喽
